crates/exex/exex/src/backfill/stream.rs

```diff
@@ -0,0 +1,99 @@
+use super::job::SingleBlockBackfillJob;
+use futures::{
+    stream::{FuturesOrdered, Stream},
+    StreamExt,
+};
+use reth_evm::execute::{BlockExecutionError, BlockExecutionOutput, BlockExecutorProvider};
+use reth_primitives::{BlockNumber, BlockWithSenders, Receipt};
+use reth_provider::{BlockReader, HeaderProvider, StateProviderFactory};
+use std::{
+    ops::RangeInclusive,
+    pin::Pin,
+    task::{ready, Context, Poll},
+};
+use tokio::task::JoinHandle;
+
+type BackfillTasks = FuturesOrdered<
+    JoinHandle<Result<(BlockWithSenders, BlockExecutionOutput<Receipt>), BlockExecutionError>>,
+>;
+
+/// The default capacity of active tasks for [`BackFillJobStream`].
+const DEFAULT_CAPACITY: usize = 100;
```

Comment:
this is quite a lot and problematic because execution is single threaded CPU bound, we should default to smth like 4 but also make this configurable